app/controllers/reviews_controller.rb

```diff
@@ -1,6 +1,23 @@
 require "mailgun"
 
 class ReviewsController < ApplicationController
+  before_action :authenticate_user!
+  before_action :require_access, only: [:edit, :destroy]
+
+  def require_access
```

Comment:
See previous comment - also, if this logic is repeated, consider adding to the `ApplicationController`, or making a new `AuthorizationController` to hold this logic, instead of repeating it.